realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.kt

```diff
@@ -681,6 +682,10 @@ class ClassMetaData(env: ProcessingEnvironment, typeMirrors: TypeMirrors, privat
             }
         }
 
+        if (Utils.isMixed(field)) {
+            Utils.error(String.format(Locale.US, "Mixed field \"${field}\" cannot be @Required or @NotNull."))
+        }
```

Comment:
You need to return false here, to indicate an error happened. Otherwise, it might risk returning true by accident which will cause the annotation processor to continue instead of aborting.